### ISSUE TITLE
Ferry routes rendering. Differentiate between car and pedestrian only routes.

### ISSUE THREAD

**Issue by Gazer75** [gravitystorm/openstreetmap-carto#1784]
Any chance this can be done? Maybe by different colour?

With more and more routes being added on the Norwegian coast its getting very crowded and hard to distinguish between the car ferries and the express boats that have no vehicle capacity.

"Express boats" "should" be tagged with motor_vehicle=no, but no idea if that is being done. Some people say it is not needed as these routes should not be directly connected to a road, but rather to a footway or path.

**Comment by matkoniecz** [comment]
> but no idea if that is being done

http://overpass-turbo.eu/ is a good tool to check things like that

**Comment by matkoniecz** [comment]
> "Express boats" "should" be tagged with motor_vehicle=no

Where it is documented? I found http://wiki.openstreetmap.org/wiki/Proposed_features/ferry#cruise_ships.2C_cruise_ferries.2C_express_boats that proposes to use ferry key.

> Any chance this can be done? Maybe by different colour?

I have no ideas for rendering that would be usable.

> Some people say it is not needed as these routes should not be directly connected to a road, but rather to a footway or path.

Maybe it is simply clear from context and there is no need for special rendering? Can you give example of location where such change would improve map?

**Comment by pnorman** [comment]
> > Some people say it is not needed as these routes should not be directly connected to a road, but rather to a footway or path.
> Maybe it is simply clear from context and there is no need for special rendering? Can you give example of location where such change would improve map?

A ferry terminal could have one route leaving it which is passenger only and another which allows vehicles. Locally, I know of some examples where the far terminals for two routes originating from the same place support different modes of transport, so the ferry routes support different modes.

**Comment by Gazer75** [comment]
@matkoniecz In this area: https://www.openstreetmap.org/#map=10/59.9509/5.4547 it is not very clear what route is doing what by just looking at it. You'll have to zoom in to each dock to see if the route is connected to a footway or not.
In this area there are at least 5 car ferries and even more express boats crossing paths.

I am aware that express boat tagging is nothing but a proposal at this point, but it would be nice to have some proper tagging for each kind of ferry. I guess Norway is a special case, at least in Europe, with our jagged coast and lots of ferry routes.

We have the big full blown car ferries that usually have a capacity of around 30-200+ car units. Then there is the smaller ferries with a small aft deck with space for 2-3 cars, and finally the regular no vehicle express boats that run quite fast. Though the modern gas ferries running on the two E39 crossings between Bergen and Stavanger also have a high speed.

The small combi ferries are mostly used in the Florø area: https://www.openstreetmap.org/#map=11/61.6080/4.9672 People living on the small islands have cars and tractors :)
Note that only the south route is currently fully mapped here.
In this area there is also an express boat going between Bergen and Selje that docks at Florø on its way, but as you can see there is no way to quickly see which route can actually carry a vehicle.

Note that there are many express boat routes not yet mapped along the coast that would cross path with car ferries.

**Comment by matkoniecz** [comment]
>  it would be nice to have some proper tagging for each kind of ferry

Proposal for new tagging should be formulated/made on OSM wiki and tagging mailing list rather than here.

**Comment by pnorman** [comment]
> > Any chance this can be done? Maybe by different colour?
> I have no ideas for rendering that would be usable.

Neither do I. @Gazer75, did you have any suggestions? I live in BC, which has its share of ferries, and I think local practice would to be to append (foot) to the name on the map, but that's not a good option for a multi-lingual map. I just can't think of any way to make it evident the differences with dasharrays/etc.

**Comment by Gazer75** [comment]
The proposal is 4 years old, but no comments. I believe have added at least motor_vehicle=no to many routes that only do passengers. Would be a start to simply differentiate using that tag?

Like it or not, but many do not bother to tag or make special tag if no map actually show a difference :)
If one map does then maybe others will follow ;)

**Comment by Gazer75** [comment]
@pnorman Maybe a different coloring of the route? Footways are red, maybe red for passengers only routes?

**Comment by imagico** [comment]
For reference - there are currently 817 ways with route=ferry and motor_vehicle=no.  A large part of this is inland, like lake ferries where motor_vehicle=no is frequently obvious.  It would make sense to use this, esp. to display these at a later zoom level to avoid clutter (z7 is way too early for many smaller routes).

**Comment by BalooUriza** [comment]
Could render it similarly, based on access tagging, to American conventions, which often use a style highly similar to osm-carto for rendering ferries.  A typical example would be similar to maps of the Puget Sound area, which might render a line as "M.V. Coho (Black Ball Transport, truck, bus, bicycle, pedestrian, kayak)" or "Victoria Express II (Black Ball Transport, pedestrian)".  Such an example might render my favorite ferry with the label "Canby Ferry (Clackamas County Transportation Maintenance, hgv, goods, bus, bicycle, foot)" and a more local example as "Oklahoma City Public Transit (Oklahoma River Cruises, foot)"

**Comment by pnorman** [comment]
> which might render a line as "M.V. Coho (Black Ball Transport, truck, bus, bicycle, pedestrian, kayak)"

This is not a good option for a multi-lingual map.

**Comment by Gazer75** [comment]
After thinking long and hard about this I realize its going to be very hard to get this working properly.
Some areas have so many routes that you have to combine the lines and use relations to not clutter the map. This means some routes can have vehicles and and some not on the same ferry way.
So unless we make combo lines for the ways that have both by reading relation data it will be difficult.

Very good example is here: https://www.openstreetmap.org/#map=12/61.6044/4.9360
Have now completed most of the ferry routes, but due to the complexity of the routes here its very hard.
Most docks are only hailed by preorder. And to make it more complicated only some of the ports have a car bridge. Basically there are two small combo ferries running all the routes.
One run the north routes between Kinn and Hovden islands, but here only the west 3 ports have car bridges: west on Reksta (Rognaldsvåg hamlet), Fanøy , Hovden (Barekstad hamlet) and Villevik (at secondary road 577 north of Florø), the rest are just passengers.

On the south route there are only two car bridges, at Askrova and Svanøya, the rest are passengers only.

Basically only the docks for cars have schedules, the rest are only used if passengers.
Still need to call and check for / order a spot if you bring a car due to the limited, 2-3, number of spots ;)

The best thing that can be done is render pure vehicle ferries with a unique render depending on the road class. So if the tag says ferry=trunk it would be some kind of green dashed line, but wider than today. This would at least make the these kinds of ferries stand out among all the other crossing express boats.

Sorry for the very long post, but felt I had to explain the spaghetti of routes :)

**Comment by matkoniecz** [comment]
Given that nobody has a good idea how it should be displayed to differentiate between these two situations I will close this issue.

New ideas about better rendering in this situation are welcomed.

Also, iff 

> "Express boats" "should" be tagged with motor_vehicle=no

http://wiki.openstreetmap.org/wiki/Proposed_features/ferry#cruise_ships.2C_cruise_ferries.2C_express_boats proposal to use ferry key is replaced by different tagging method wiki should be updated.

**Comment by Gazer75** [comment]
I guess the issue is pretty rare world wide for people to care about this.

I already suggested using a dashed line similar to the road it represent. As the proposal suggests one would use ferry=trunk and so on for regular vehicle ferries. This would make it very easy to set them apart.
There is no need to change the proposal at all. By using the ferry= it will be easy to differentiate ferry routes.

**Comment by matkoniecz** [comment]
> I already suggested using a dashed line similar to the road it represent.

Sorry, I missed it.

**Comment by Gazer75** [comment]
I am no graphics artist so can not make visual suggestions here. But the current spacing and length of the dashes on ferry routes is good, just make them as wide as a regular highway and the same color. Depending on class used in the ferry= tag.

**Comment by BalooUriza** [comment]
On Tue, Sep 15, 2015 at 6:39 AM, Gazer75 <notifications@github.com> wrote:

> I guess the issue is pretty rare world wide for people to care about this.
>
> I already suggested using a dashed line similar to the road it represent.
> As the proposal suggests one would use ferry=trunk and so on for regular
> vehicle ferries. This would make it very easy to set them apart.
> There is no need to change the proposal at all. By using the ferry= it
> will be easy to differentiate ferry routes.
>
I missed that suggestion, but I like it.  Not sure how often a trunk might
cross a ferry (outside Japan's funky tagging scheme which I won't touch
with a barge pole due to being illiterate in the language and having
literally no idea how to comprehend that culture's rather (from my
perspective, incomprehensibly counterintuitive)
paperwork-rules-everything-around-me attitude towards cartography), but
could definitely see a primary or smaller crossing for the ferries I'm
familiar with and might lend a hand in mapping.  I do see an issue with the
nonmotorized modes, though, as the dotted and dashed lines might be
confusing (since ferry=path or ferry=cycleway; foot=yes are likely taggings
for such a route).

This discussion might need to get tossed to tagging@l.o.o for further
refinement.

**Comment by Gazer75** [comment]
There are quite a few ferries in Norway that are trunk roads. Here all E routes and national highways are set as trunks, and we have a lot of ferry crossings along the coast. on E39 along the west coast, between Stavanger and Trondheim, there are 7 ferry crossings I believe. For other national roads tagged trunk I found 8. On E6 there is one crossing between Bodø and Narvik.

For primary and secondary roads I have no idea how many crossings there are, but a search in a database showed 332 terminals connected to such roads, so there are at least 150 crossings.

**Comment by pnorman** [comment]
checking ferry tags on ways, 

```
┌───────┬────────────────┐
│ count │     ferry      │
├───────┼────────────────┤
│   275 │ yes            │
│   185 │ tertiary       │
│   139 │ secondary      │
│   133 │ unclassified   │
│   107 │ primary        │
│    45 │ trunk          │
│    29 │ tourist        │
│    18 │ local          │
│    13 │ residential    │
│     7 │ footway        │
│     6 │ track          │
│     6 │ pedestrian     │
```
This isn't wide enough usage for us to make use of it. There is no established concept of a trunk, primary, secondary, etc ferry in OSM.

But, that's not what this issue is about, which is rendering transportation type access restrictions. If you want to support `ferry=*`, a new issue should be used.

Looking at the stats

![image](https://cloud.githubusercontent.com/assets/1190866/9891686/fa7bec8e-5bbc-11e5-98db-bebd240a4604.png)

The distinction we need to worry about is ferries which allow vehicles, don't allow vehicles, or have an unknown state, probably collapsing unknown into allow.

**Comment by Gazer75** [comment]
As I've said before, most people do not add tags that are not used by a renderer.
The proposal is old and was never voted on, but some still used the tags.

Just because there are not thousands of ferry=* doesn't mean it can't be used.

Also do not see why a new issue should be made as it is essentially the same. To not render all ferries the same.

**Comment by Janjko** [comment]
>Access rendering is not going to be possible as ferry routes can be built with route relations and thus a ferry way can have multiple routes with different restrictions.

I'm not sure if our carto stack supports it, but I wouldn't call it impossible. Just follow these rules:

1. If the way has motorcar=yes then it supports vehicles
2. If the way is in a route that has motorcar=yes, than it supports vehicles
3. If 1 or 2 are not met, it only supports passengers

And instead of thickness, I would rather have icons on routes, just like shields on roads, something like this: [passenger route](http://www2.psd100.com/ppp/2013/11/0501/ship-icon-1105173535.png) and [vehicle route](https://cdn2.iconfinder.com/data/icons/windows-8-metro-style/512/ferry-.png)

**Comment by jeisenbe** [comment]
Currently there are 2333 `route=ferry` ways with `motor_vehicle=no`, out of 21k ferry route ways, so that's over 10% - a significant percentage. There are another 904 with `motorcar=no` which could be treated the same, perhaps (http://overpass-turbo.eu/s/NWS).
